plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/ImplementCountWithDistinct.java

```diff
@@ -0,0 +1,90 @@
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc.expression;
+
+import io.trino.matching.Capture;
+import io.trino.matching.Captures;
+import io.trino.matching.Pattern;
+import io.trino.plugin.jdbc.JdbcClient;
+import io.trino.plugin.jdbc.JdbcColumnHandle;
+import io.trino.plugin.jdbc.JdbcExpression;
+import io.trino.plugin.jdbc.JdbcTypeHandle;
+import io.trino.spi.connector.AggregateFunction;
+import io.trino.spi.expression.Variable;
+import io.trino.spi.type.BigintType;
+import io.trino.spi.type.CharType;
+import io.trino.spi.type.VarcharType;
+
+import java.util.Optional;
+
+import static com.google.common.base.Verify.verify;
+import static io.trino.matching.Capture.newCapture;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.functionName;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.hasFilter;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.hasSortOrder;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.singleInput;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.variable;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Implements {@code count([DISTINCT] x)}.
+ */
+public class ImplementCountWithDistinct
+        implements AggregateFunctionRule
+{
+    private static final Capture<Variable> INPUT = newCapture();
+
+    private final JdbcTypeHandle bigintTypeHandle;
+    private final boolean isRemoteCollationSensitive;
+
+    /**
+     * @param bigintTypeHandle A {@link JdbcTypeHandle} that will be mapped to {@link BigintType} by {@link JdbcClient#toColumnMapping}.
+     */
+    public ImplementCountWithDistinct(JdbcTypeHandle bigintTypeHandle, boolean isRemoteCollationSensitive)
+    {
+        this.bigintTypeHandle = requireNonNull(bigintTypeHandle, "bigintTypeHandle is null");
+        this.isRemoteCollationSensitive = isRemoteCollationSensitive;
+    }
+
+    @Override
+    public Pattern<AggregateFunction> getPattern()
+    {
+        return Pattern.typeOf(AggregateFunction.class)
+                .with(hasSortOrder().equalTo(false))
+                .with(hasFilter().equalTo(false))
+                .with(functionName().equalTo("count"))
+                .with(singleInput().matching(variable().capturedAs(INPUT)));
+    }
+
+    @Override
+    public Optional<JdbcExpression> rewrite(AggregateFunction aggregateFunction, Captures captures, RewriteContext context)
+    {
+        Variable input = captures.get(INPUT);
+        JdbcColumnHandle columnHandle = (JdbcColumnHandle) context.getAssignment(input.getName());
+        String distinct = aggregateFunction.isDistinct() ? "DISTINCT " : "";
+        verify(aggregateFunction.getOutputType() == BIGINT);
+
+        // Remote database is case insensitive or compares values differently from Trino
```

Comment:
the comment is off -- we don't know yet.

maybe move the comment inside the `if` block